torch/distributed/tensor/parallel/style.py

```diff
@@ -179,10 +179,6 @@ def __init__(
         super().__init__()
         self.input_layouts = (input_layouts or Shard(-1), )
```

Comment:
Now we have two use cases nn.Linear and nn.Embedding, with two different `desired_input_layouts`. In the long term does it still make sense to keep default `input_layouts` to be `Shard(-1)`? This seems favoring nn.Linear use cases. In the short term, we need to keep it for backward compatibility.

Reply:
That's a good question. Yeah for BC reasons I think we probably can't change the default input layouts for rowwise parallel, and I think we should indeed favoring nn.Linear instead of nn.Embedding as Embedding only happen in the first transformer layer most of time.

I think currently user have to pass `input_layouts=Replicate()` for models like Transoformer, one thing we can do is that maybe we should add a warning if we found that we are running row sharded embedding and warn user if the input_layouts is not replicate?